### ISSUE TITLE
Key presses with matching timestamps are being discarded on X11

### ISSUE THREAD

**Issue by zlsa** [glfw/glfw#1415]
# OS and version
`Linux [hostname] 4.18.0-13-generic #14-Ubuntu SMP Wed Dec 5 09:04:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

Commit c90c7b97109db909577e3bf540b5f884422b7e14

No error messages.

# Output of `tests/events`

In the below log, I simultaneously pressed left control and left shift, then pressed J after a short delay. Shift and control are both marked in the modifiers bitmask even though a press event was never received for the left control press.

```
Library initialized
Creating windowed mode window 1 (640x480)
Main loop starting
00000000 to 1 at 0.234: Window position: 1600 858
00000001 to 1 at 0.234: Window refresh
00000002 to 1 at 0.235: Window focused
00000003 to 1 at 0.963: Key 0x0154 Scancode 0x0032 (LEFT SHIFT) (with no mods) was pressed
00000004 to 1 at 2.746: Key 0x004a Scancode 0x002c (J) (j) (with shift control) was pressed
00000005 to 1 at 2.866: Key 0x004a Scancode 0x002c (J) (j) (with shift control) was released
00000006 to 1 at 3.659: Key 0x0154 Scancode 0x0032 (LEFT SHIFT) (with shift control) was released
0000000c to 1 at 5.363: Key 0x0156 Scancode 0x0040 (LEFT ALT) (with no mods) was pressed
0000000d to 1 at 5.497: Window close
```

See also 4ff66a7818e3eaa5362c828a18220b12f1cd9bc8, #747, and #964.

**Comment by zlsa** [comment]
Moving the `_glfwInputKey` call to outside of the `lastKeyTime` `if` statement in `src/x11_window.c` fixes the issue on my system:

```c
                // Line 1259
                //if (window->x11.lastKeyTime < event->xkey.time)                                                                                                                                                                                                                                                                                                                                                             
                //{                                                                                                                                                                                                                                                                                                                                                                                                                
                //    if (keycode)                                                                                                                                                                                                                                                                                                                                                                                                 
                //        _glfwInputKey(window, key, keycode, GLFW_PRESS, mods);                                                                                                                                                                                                                                                                                                                                                   
                //                                                                                                                                                                                                                                                                                                                                                                                                                 
                //    window->x11.lastKeyTime = event->xkey.time;                                                                                                                                                                                                                                                                                                                                                                  
                //}                                                                                                                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                                                                                                                                 
                _glfwInputKey(window, key, keycode, GLFW_PRESS, mods);                                                                                                                                                                                                                                                                                                                                                           
```

**Comment by zlsa** [comment]
I understand that the original issue may still be present on some systems, possibly requiring a more nuanced way of resolving this issue, but the below patch 100% solves this issue for me.

```diff
diff --git a/src/x11_window.c b/src/x11_window.c
index 5e916107..36abdf78 100644
--- a/src/x11_window.c
+++ b/src/x11_window.c
@@ -1256,13 +1256,15 @@ static void processEvent(XEvent *event)
                 //       These have the same timestamp as the original event
                 //       Corresponding release events are filtered out
                 //       implicitly by the GLFW key repeat logic
-                if (window->x11.lastKeyTime < event->xkey.time)
-                {
-                    if (keycode)
-                        _glfwInputKey(window, key, keycode, GLFW_PRESS, mods);
 
-                    window->x11.lastKeyTime = event->xkey.time;
-                }
+                //if (window->x11.lastKeyTime < event->xkey.time)
+                //{
+                //    if (keycode)
+                //        _glfwInputKey(window, key, keycode, GLFW_PRESS, mods);
+                //
+                //    window->x11.lastKeyTime = event->xkey.time;
+                //}
+                _glfwInputKey(window, key, keycode, GLFW_PRESS, mods);
 
                 if (!filtered)
                 {
```

**Comment by elmindreda** [comment]
Thank you for reporting this. It should be fixed now with 9a3664b4a992a5a9b669179098542b0f882dece7.
